lib/paramsFromUrl.ts

```diff
@@ -2,15 +2,14 @@ import { getValidUrlFromInput } from './getValidUrlFromInput';
 import { slugsFromUrl } from './slugsFromUrl';
 import { UrlMatchSlugs } from './types';
 
-export const paramsFromUrl = (urlString: string): UrlMatchSlugs => {
-
+export function paramsFromUrl (urlString: string): UrlMatchSlugs | undefined {
   try {
     const urlObj = getValidUrlFromInput(urlString)
     const matchResult = slugsFromUrl(urlObj.pathname)
     if (matchResult !== false && matchResult.params != null) {
       return { ...matchResult.params };
     }
-  } catch { }
-
-  throw new Error('Unsupported URL');
+  } catch {
+    throw new Error('Unsupported URL');
+  }
 };
```

Comment:
👎 
This was intentional. We should not return undefined from this function. Overloading types is one of the major reason why this codebase is so painful. We should use strict return types and throw errors if that isn't the case. Consumers should handle those errors and act accordingly.

Reply:
if it makes you more comfortable, we can wrap it in a `finally` block?